File: day1/ruby/part_1.rb
input_file = "input.txt"

left_list = []
right_list = []

File.foreach(input_file) do |line|
  left, right = line.split.map(&:to_i)
  left_list << left
  right_list << right
end

left_list.sort!
right_list.sort!

total_distance = left_list.each_with_index.sum { |left, i| (left - right_list[i]).abs }

puts total_distance

File: day1/typescript/part_1.ts
import * as fs from 'fs';

const inputFile = "input.txt";

const fileContent = fs.readFileSync(inputFile, 'utf-8');
const lines = fileContent.trim().split('\n');

const leftList: number[] = [];
const rightList: number[] = [];

lines.forEach(line => {
    const [left, right] = line.split(' ').map(Number);
    leftList.push(left);
    rightList.push(right);
});

leftList.sort((a, b) => a - b);
rightList.sort((a, b) => a - b);

const totalDistance = leftList.reduce((sum, left, i) => sum + Math.abs(left - rightList[i]), 0);

console.log(totalDistance);
